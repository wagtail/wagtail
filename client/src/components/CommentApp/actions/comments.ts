import type {
  Comment,
  CommentUpdate,
  CommentReply,
  CommentReplyUpdate,
} from '../state/comments';

export const ADD_COMMENT = 'add-comment';
export const UPDATE_COMMENT = 'update-comment';
export const DELETE_COMMENT = 'delete-comment';
export const SET_FOCUSED_COMMENT = 'set-focused-comment';
export const SET_PINNED_COMMENT = 'set-pinned-comment';

export const ADD_REPLY = 'add-reply';
export const UPDATE_REPLY = 'update-reply';
export const DELETE_REPLY = 'delete-reply';

export const INVALIDATE_CONTENT_PATH = 'invalidate-content-path';

export interface AddCommentAction {
  type: typeof ADD_COMMENT;
  comment: Comment;
}

export interface UpdateCommentAction {
  type: typeof UPDATE_COMMENT;
  commentId: number;
  update: CommentUpdate;
}

export interface DeleteCommentAction {
  type: typeof DELETE_COMMENT;
  commentId: number;
}

export interface SetFocusedCommentAction {
  type: typeof SET_FOCUSED_COMMENT;
  commentId: number | null;
  updatePinnedComment: boolean;
}

export interface AddReplyAction {
  type: typeof ADD_REPLY;
  commentId: number;
  reply: CommentReply;
}

export interface UpdateReplyAction {
  type: typeof UPDATE_REPLY;
  commentId: number;
  replyId: number;
  update: CommentReplyUpdate;
}

export interface DeleteReplyAction {
  type: typeof DELETE_REPLY;
  commentId: number;
  replyId: number;
}

export interface InvalidateContentPathAction {
  type: typeof INVALIDATE_CONTENT_PATH;
  contentPath: string;
}

export type Action =
  | AddCommentAction
  | UpdateCommentAction
  | DeleteCommentAction
  | SetFocusedCommentAction
  | AddReplyAction
  | UpdateReplyAction
  | DeleteReplyAction
  | InvalidateContentPathAction;

export function addComment(comment: Comment): AddCommentAction {
  return {
    type: ADD_COMMENT,
    comment,
  };
}

export function updateComment(
  commentId: number,
  update: CommentUpdate
): UpdateCommentAction {
  return {
    type: UPDATE_COMMENT,
    commentId,
    update,
  };
}

export function deleteComment(commentId: number): DeleteCommentAction {
  return {
    type: DELETE_COMMENT,
    commentId,
  };
}

export function setFocusedComment(
  commentId: number | null,
  { updatePinnedComment } = { updatePinnedComment: false }
): SetFocusedCommentAction {
  return {
    type: SET_FOCUSED_COMMENT,
    commentId,
    updatePinnedComment
  };
}

export function addReply(
  commentId: number,
  reply: CommentReply
): AddReplyAction {
  return {
    type: ADD_REPLY,
    commentId,
    reply,
  };
}

export function updateReply(
  commentId: number,
  replyId: number,
  update: CommentReplyUpdate
): UpdateReplyAction {
  return {
    type: UPDATE_REPLY,
    commentId,
    replyId,
    update,
  };
}

export function deleteReply(
  commentId: number,
  replyId: number
): DeleteReplyAction {
  return {
    type: DELETE_REPLY,
    commentId,
    replyId,
  };
}

export function invalidateContentPath(contentPath: string): InvalidateContentPathAction {
  return {
    type: INVALIDATE_CONTENT_PATH,
    contentPath,
  };
}

export const commentActionFunctions = {
  addComment,
  updateComment,
  deleteComment,
  setFocusedComment,
  addReply,
  updateReply,
  deleteReply,
  invalidateContentPath,
};
