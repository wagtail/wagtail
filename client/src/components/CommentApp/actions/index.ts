import type { Action as CommentsAction } from './comments';
import type { Action as SettingsActon } from './settings';

export type Action = CommentsAction | SettingsActon;
