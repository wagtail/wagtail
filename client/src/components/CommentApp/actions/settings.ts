import type { SettingsStateUpdate } from '../state/settings';

export const UPDATE_GLOBAL_SETTINGS = 'update-global-settings';

export interface UpdateGlobalSettingsAction {
  type: typeof UPDATE_GLOBAL_SETTINGS;
  update: SettingsStateUpdate;
}

export type Action = UpdateGlobalSettingsAction;

export function updateGlobalSettings(
  update: SettingsStateUpdate,
): UpdateGlobalSettingsAction {
  return {
    type: UPDATE_GLOBAL_SETTINGS,
    update,
  };
}
