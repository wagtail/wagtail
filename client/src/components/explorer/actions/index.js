import { createAction } from 'redux-actions';

import { API_PAGES, PAGES_ROOT_ID } from 'config';

function _getHeaders() {
  const headers = new Headers();
  headers.append('Content-Type', 'application/json');

  return {
    credentials: 'same-origin',
    headers: headers,
    method: 'GET'
  };
}

function _get(url) {
  return fetch(url, _getHeaders()).then(response => response.json());
}

export const fetchStart = createAction('FETCH_START');

export const fetchSuccess = createAction('FETCH_SUCCESS', (id, body) => {
  return { id, body };
});

export const fetchFailure = createAction('FETCH_FAILURE');

export const pushPage = createAction('PUSH_PAGE');

export const popPage = createAction('POP_PAGE');

export const fetchBranchSuccess = createAction('FETCH_BRANCH_SUCCESS', (id, json) => {
  return { id, json };
});

export const fetchBranchStart = createAction('FETCH_BRANCH_START');

export const clearError = createAction('CLEAR_TRANSPORT_ERROR');

export const resetTree = createAction('RESET_TREE');

export const treeResolved = createAction('TREE_RESOLVED');

// Make this a bit better... hmm....
export function fetchTree(id = 1) {
  return (dispatch) => {
    dispatch(fetchBranchStart(id));

    return _get(`${API_PAGES}${id}/`)
      .then(json => {
        dispatch(fetchBranchSuccess(id, json));

        // Recursively walk up the tree to the root, to figure out how deep
        // in the tree we are.
        if (json.meta.parent) {
          dispatch(fetchTree(json.meta.parent.id));
        } else {
          dispatch(treeResolved());
        }
      });
  };
}

export function fetchRoot() {
  return (dispatch) => {
    // TODO Should not need an id.
    dispatch(resetTree(1));

    return _get(`${API_PAGES}?child_of=${PAGES_ROOT_ID}`)
      .then(json => {
        // TODO right now, only works for a single homepage.
        // TODO What do we do if there is no homepage?
        const rootId = json.items[0].id;

        dispatch(fetchTree(rootId));
      });
  };
}

export const toggleExplorer = createAction('TOGGLE_EXPLORER');

export const fetchChildrenSuccess = createAction('FETCH_CHILDREN_SUCCESS', (id, json) => {
  return { id, json };
});

export const fetchChildrenStart = createAction('FETCH_CHILDREN_START');

/**
 * Gets the children of a node from the API
 */
export function fetchChildren(id = 'root') {
  return (dispatch, getState) => {
    const { explorer } = getState();

    let api = `${API_PAGES}?child_of=${id}`;

    if (explorer.fields) {
      api += `&fields=${explorer.fields.map(global.encodeURIComponent).join(',')}`;
    }

    if (explorer.filter) {
      api = `${api}&${explorer.filter}`;
    }

    dispatch(fetchChildrenStart(id));

    return _get(api)
      .then(json => dispatch(fetchChildrenSuccess(id, json)));
  };
}

export function setFilter(filter) {
  return (dispatch, getState) => {
    const { explorer } = getState();
    const id = explorer.path[explorer.path.length - 1];

    dispatch({
      payload: {
        filter,
        id
      },
      type: 'SET_FILTER'
    });

    dispatch(fetchChildren(id));
  };
}

/**
 * TODO: determine if page is already loaded, don't load it again, just push.
 */
export function fetchPage(id = 1) {
  return dispatch => {
    dispatch(fetchStart(id));
    return _get(`${API_PAGES}${id}/`)
      .then(json => dispatch(fetchSuccess(id, json)))
      .then(json => dispatch(fetchChildren(id, json)))
      .catch(json => dispatch(fetchFailure(new Error(JSON.stringify(json)))));
  };
}

export const setDefaultPage = createAction('SET_DEFAULT_PAGE');
