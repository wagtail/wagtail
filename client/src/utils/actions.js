const identity = func => func;

// Stolen from the following project (had a 18kb footprint at the time).
// https://github.com/acdlite/redux-actions/blob/79c68635fb1524c1b1cf8e2398d4b099b53ca8de/src/createAction.js
export function createAction(type, actionCreator, metaCreator) {
  const finalActionCreator = typeof actionCreator === 'function' ? actionCreator : identity;

  return (...args) => {
    const action = {
      type,
      payload: finalActionCreator(...args),
    };

    if (action.payload instanceof Error) {
      // Handle FSA errors where the payload is an Error object. Set error.
      action.error = true;
    }

    if (typeof metaCreator === 'function') {
      action.meta = metaCreator(...args);
    }

    return action;
  };
}
