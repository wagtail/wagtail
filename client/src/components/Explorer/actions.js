import * as admin from '../../api/admin';
import { createAction } from '../../utils/actions';
import { MAX_EXPLORER_PAGES } from '../../config/wagtailConfig';

const getPageStart = createAction('GET_PAGE_START');
const getPageSuccess = createAction('GET_PAGE_SUCCESS', (id, data) => ({ id, data }));
const getPageFailure = createAction('GET_PAGE_FAILURE', (id, error) => ({ id, error }));

/**
 * Gets a page from the API.
 */
function getPage(id) {
  return (dispatch) => {
    dispatch(getPageStart(id));

    return admin.getPage(id).then((data) => {
      dispatch(getPageSuccess(id, data));
    }, (error) => {
      dispatch(getPageFailure(id, error));
    });
  };
}

const getChildrenStart = createAction('GET_CHILDREN_START', id => ({ id }));
const getChildrenSuccess = createAction('GET_CHILDREN_SUCCESS', (id, items, meta) => ({ id, items, meta }));
const getChildrenFailure = createAction('GET_CHILDREN_FAILURE', (id, error) => ({ id, error }));

/**
 * Gets the children of a node from the API.
 */
function getChildren(id, offset = 0) {
  return (dispatch) => {
    dispatch(getChildrenStart(id));

    return admin.getPageChildren(id, {
      offset: offset,
    }).then(({ items, meta }) => {
      const nbPages = offset + items.length;
      dispatch(getChildrenSuccess(id, items, meta));

      // Load more pages if necessary. Only one request is created even though
      // more might be needed, thus naturally throttling the loading.
      if (nbPages < meta.total_count && nbPages < MAX_EXPLORER_PAGES) {
        dispatch(getChildren(id, nbPages));
      }
    }, (error) => {
      dispatch(getChildrenFailure(id, error));
    });
  };
}

const openExplorer = createAction('OPEN_EXPLORER', id => ({ id }));
export const closeExplorer = createAction('CLOSE_EXPLORER');

export function toggleExplorer(id) {
  return (dispatch, getState) => {
    const { explorer, nodes } = getState();

    if (explorer.isVisible) {
      dispatch(closeExplorer());
    } else {
      const page = nodes[id];

      dispatch(openExplorer(id));

      if (!page) {
        dispatch(getChildren(id));
      }

      // We need to get the title of the starting page, only if it is not the site's root.
      const isNotRoot = id !== 1;
      if (isNotRoot) {
        dispatch(getPage(id));
      }
    }
  };
}

export const popPage = createAction('POP_PAGE');
const pushPagePrivate = createAction('PUSH_PAGE', id => ({ id }));

export function pushPage(id) {
  return (dispatch, getState) => {
    const { nodes } = getState();
    const page = nodes[id];

    dispatch(pushPagePrivate(id));

    if (page && !page.isFetching && !page.children.count > 0) {
      dispatch(getChildren(id));
    }
  };
}
