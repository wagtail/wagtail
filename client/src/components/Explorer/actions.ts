import { ThunkAction } from 'redux-thunk';

import * as admin from '../../api/admin';
import { createAction } from '../../utils/actions';
import { MAX_EXPLORER_PAGES } from '../../config/wagtailConfig';

import { State as ExplorerState, Action as ExplorerAction } from './reducers/explorer';
import { State as NodeState, WagtailPageAPI, Action as NodeAction } from './reducers/nodes';

interface State {
  explorer: ExplorerState,
  nodes: NodeState,
}

type Action = ExplorerAction | NodeAction;
type ThunkActionType = ThunkAction<void, State, unknown, Action>;

const getPageSuccess = createAction('GET_PAGE_SUCCESS', (id: number, data: WagtailPageAPI) => ({ id, data }));
const getPageFailure = createAction('GET_PAGE_FAILURE', (id: number, error: Error) => ({ id, error }));

/**
 * Gets a page from the API.
 */
function getPage(id: number): ThunkActionType {
  return (dispatch) => admin.getPage(id).then((data) => {
    dispatch(getPageSuccess(id, data));
  }, (error) => {
    dispatch(getPageFailure(id, error));
  });
}

const getChildrenStart = createAction('GET_CHILDREN_START', (id: number) => ({ id }));
const getChildrenSuccess = createAction(
  'GET_CHILDREN_SUCCESS',
  (id, items: WagtailPageAPI[], meta: any) => ({ id, items, meta })
);
const getChildrenFailure = createAction('GET_CHILDREN_FAILURE', (id: number, error: Error) => ({ id, error }));

/**
 * Gets the children of a node from the API.
 */
function getChildren(id: number, offset = 0): ThunkActionType {
  return (dispatch) => {
    dispatch(getChildrenStart(id));

    return admin.getPageChildren(id, {
      offset: offset,
    }).then(({ items, meta }) => {
      const nbPages = offset + items.length;
      dispatch(getChildrenSuccess(id, items, meta));

      // Load more pages if necessary. Only one request is created even though
      // more might be needed, thus naturally throttling the loading.
      if (nbPages < meta.total_count && nbPages < MAX_EXPLORER_PAGES) {
        dispatch(getChildren(id, nbPages));
      }
    }, (error) => {
      dispatch(getChildrenFailure(id, error));
    });
  };
}

const openExplorer = createAction('OPEN_EXPLORER', id => ({ id }));
export const closeExplorer = createAction('CLOSE_EXPLORER');

export function toggleExplorer(id: number): ThunkActionType {
  return (dispatch, getState) => {
    const { explorer, nodes } = getState();

    if (explorer.isVisible) {
      dispatch(closeExplorer());
    } else {
      const page = nodes[id];

      dispatch(openExplorer(id));

      if (!page) {
        dispatch(getChildren(id));
      }

      // We need to get the title of the starting page, only if it is not the site's root.
      const isNotRoot = id !== 1;
      if (isNotRoot) {
        dispatch(getPage(id));
      }
    }
  };
}

export const popPage = createAction('POP_PAGE');
const pushPagePrivate = createAction('PUSH_PAGE', (id: number) => ({ id }));

export function pushPage(id: number): ThunkActionType {
  return (dispatch, getState) => {
    const { nodes } = getState();
    const page = nodes[id];

    dispatch(pushPagePrivate(id));

    if (page && !page.isFetching && !(page.children.count > 0)) {
      dispatch(getChildren(id));
    }
  };
}
