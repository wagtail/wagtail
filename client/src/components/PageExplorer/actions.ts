import { ThunkAction } from 'redux-thunk';

import * as admin from '../../api/admin';
import { createAction } from '../../utils/actions';
import { MAX_EXPLORER_PAGES } from '../../config/wagtailConfig';

import { State, Action } from './reducers';

type ThunkActionType = ThunkAction<void, State, unknown, Action>;

const getPageSuccess = createAction(
  'GET_PAGE_SUCCESS',
  (id: number, data: admin.WagtailPageAPI) => ({ id, data }),
);
const getPageFailure = createAction(
  'GET_PAGE_FAILURE',
  (id: number, error: Error) => ({ id, error }),
);

/**
 * Gets a page from the API.
 */
function getPage(id: number): ThunkActionType {
  return (dispatch) =>
    admin.getPage(id).then(
      (data) => {
        dispatch(getPageSuccess(id, data));
      },
      (error) => {
        dispatch(getPageFailure(id, error));
      },
    );
}

const getChildrenStart = createAction('GET_CHILDREN_START', (id: number) => ({
  id,
}));
const getChildrenSuccess = createAction(
  'GET_CHILDREN_SUCCESS',
  (id, items: admin.WagtailPageAPI[], meta: any) => ({ id, items, meta }),
);
const getChildrenFailure = createAction(
  'GET_CHILDREN_FAILURE',
  (id: number, error: Error) => ({ id, error }),
);

/**
 * Gets the children of a node from the API.
 */
function getChildren(id: number, offset = 0): ThunkActionType {
  return (dispatch) => {
    dispatch(getChildrenStart(id));

    return admin
      .getPageChildren(id, {
        offset: offset,
      })
      .then(
        ({ items, meta }) => {
          const nbPages = offset + items.length;
          dispatch(getChildrenSuccess(id, items, meta));

          // Load more pages if necessary. Only one request is created even though
          // more might be needed, thus naturally throttling the loading.
          if (nbPages < meta.total_count && nbPages < MAX_EXPLORER_PAGES) {
            dispatch(getChildren(id, nbPages));
          }
        },
        (error) => {
          dispatch(getChildrenFailure(id, error));
        },
      );
  };
}

const getTranslationsStart = createAction('GET_TRANSLATIONS_START', (id) => ({
  id,
}));
const getTranslationsSuccess = createAction(
  'GET_TRANSLATIONS_SUCCESS',
  (id, items) => ({ id, items }),
);
const getTranslationsFailure = createAction(
  'GET_TRANSLATIONS_FAILURE',
  (id, error) => ({ id, error }),
);

/**
 * Gets the translations of a node from the API.
 */
function getTranslations(id) {
  return (dispatch) => {
    dispatch(getTranslationsStart(id));

    return admin.getAllPageTranslations(id, { onlyWithChildren: true }).then(
      (items) => {
        dispatch(getTranslationsSuccess(id, items));
      },
      (error) => {
        dispatch(getTranslationsFailure(id, error));
      },
    );
  };
}

const openPageExplorerPrivate = createAction('OPEN_EXPLORER', (id) => ({ id }));
export const closePageExplorer = createAction('CLOSE_EXPLORER');

export function openPageExplorer(id: number): ThunkActionType {
  return (dispatch, getState) => {
    const { nodes } = getState();

    const page = nodes[id];

    dispatch(openPageExplorerPrivate(id));

    if (!page) {
      dispatch(getChildren(id));

      if (id !== 1) {
        dispatch(getTranslations(id));
      }
    }

    // We need to get the title of the starting page, only if it is not the site's root.
    const isNotRoot = id !== 1;
    if (isNotRoot) {
      dispatch(getPage(id));
    }
  };
}

const gotoPagePrivate = createAction(
  'GOTO_PAGE',
  (id: number, transition: number) => ({ id, transition }),
);

export function gotoPage(id: number, transition: number): ThunkActionType {
  return (dispatch, getState) => {
    const { nodes } = getState();
    const page = nodes[id];

    dispatch(gotoPagePrivate(id, transition));

    if (page && !page.isFetchingChildren && !(page.children.count > 0)) {
      dispatch(getChildren(id));
    }

    if (page && !page.isFetchingTranslations && page.translations == null) {
      dispatch(getTranslations(id));
    }
  };
}
