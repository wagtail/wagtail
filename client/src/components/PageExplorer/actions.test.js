import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import * as actions from './actions';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const stubState = {
  explorer: {
    isVisible: true,
  },
  nodes: {
    5: {
      isFetching: true,
      children: {},
    },
  },
};

describe('actions', () => {
  describe('closePageExplorer', () => {
    it('exists', () => {
      expect(actions.closePageExplorer).toBeDefined();
    });

    it('creates action', () => {
      expect(actions.closePageExplorer().type).toEqual('CLOSE_EXPLORER');
    });
  });

  describe('togglePageExplorer', () => {
    it('exists', () => {
      expect(actions.togglePageExplorer).toBeDefined();
    });

    it('close', () => {
      const store = mockStore(stubState);
      store.dispatch(actions.togglePageExplorer(5));
      expect(store.getActions()).toMatchSnapshot();
    });

    it('open', () => {
      const stub = Object.assign({}, stubState);
      stub.explorer.isVisible = false;
      const store = mockStore(stub);
      store.dispatch(actions.togglePageExplorer(5));
      expect(store.getActions()).toMatchSnapshot();
    });

    it('open first time', () => {
      const stub = { explorer: stubState.explorer, nodes: {} };
      stub.explorer.isVisible = false;
      const store = mockStore(stub);
      store.dispatch(actions.togglePageExplorer(5));
      expect(store.getActions()).toMatchSnapshot();
    });

    it('open at root', () => {
      const stub = Object.assign({}, stubState);
      stub.explorer.isVisible = false;
      const store = mockStore(stub);
      store.dispatch(actions.togglePageExplorer(1));
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe('gotoPage', () => {
    it('exists', () => {
      expect(actions.gotoPage).toBeDefined();
    });

    it('creates action', () => {
      const store = mockStore(stubState);
      store.dispatch(actions.gotoPage(5, 1));
      expect(store.getActions()).toMatchSnapshot();
    });

    it('triggers getChildren', () => {
      const stub = Object.assign({}, stubState);
      stub.nodes[5].isFetching = false;
      const store = mockStore(stub);
      store.dispatch(actions.gotoPage(5, 1));
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
