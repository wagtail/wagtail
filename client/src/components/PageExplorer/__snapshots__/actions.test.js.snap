// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions closePageExplorer close 1`] = `
Array [
  Object {
    "payload": undefined,
    "type": "CLOSE_EXPLORER",
  },
]
`;

exports[`actions gotoPage creates action 1`] = `
Array [
  Object {
    "payload": Object {
      "id": 5,
      "transition": 1,
    },
    "type": "GOTO_PAGE",
  },
  Object {
    "payload": Object {
      "id": 5,
    },
    "type": "GET_CHILDREN_START",
  },
  Object {
    "payload": Object {
      "id": 5,
    },
    "type": "GET_TRANSLATIONS_START",
  },
]
`;

exports[`actions gotoPage triggers getChildren 1`] = `
Array [
  Object {
    "payload": Object {
      "id": 5,
      "transition": 1,
    },
    "type": "GOTO_PAGE",
  },
  Object {
    "payload": Object {
      "id": 5,
    },
    "type": "GET_CHILDREN_START",
  },
  Object {
    "payload": Object {
      "id": 5,
    },
    "type": "GET_TRANSLATIONS_START",
  },
]
`;

exports[`actions openPageExplorer open 1`] = `
Array [
  Object {
    "payload": Object {
      "id": 5,
    },
    "type": "OPEN_EXPLORER",
  },
]
`;

exports[`actions openPageExplorer open at root 1`] = `
Array [
  Object {
    "payload": Object {
      "id": 1,
    },
    "type": "OPEN_EXPLORER",
  },
  Object {
    "payload": Object {
      "id": 1,
    },
    "type": "GET_CHILDREN_START",
  },
]
`;

exports[`actions openPageExplorer open first time 1`] = `
Array [
  Object {
    "payload": Object {
      "id": 5,
    },
    "type": "OPEN_EXPLORER",
  },
  Object {
    "payload": Object {
      "id": 5,
    },
    "type": "GET_CHILDREN_START",
  },
  Object {
    "payload": Object {
      "id": 5,
    },
    "type": "GET_TRANSLATIONS_START",
  },
]
`;
